src/Language/PureScript/TypeChecker/Entailment/IntCompare.hs

```diff
@@ -0,0 +1,173 @@
+-- |
+-- Graph-based solver for comparing type-level numbers with respect to
+-- reflexivity, symmetry, and transitivity properties.
+--
+module Language.PureScript.TypeChecker.Entailment.IntCompare where
+
+import Protolude
+
+import Control.Arrow ((&&&))
+import qualified Data.Graph as G
+import qualified Data.Map as M
+import qualified Data.Set as S
+
+import qualified Language.PureScript.Types as P
+import qualified Language.PureScript.Constants.Prim as P
+
+data Comparison a
+  = Equal a a
+  | LessThan a a
+  | GreaterThan a a
```

Comment:
Do we need this third constructor? Looking at your approach, it seems to me like we could always choose to represent a `GreaterThan x y` as `LessThan y x` instead, and everything would continue to work.